app/javascript/tests/close_contact/CloseContactTable.test.js

```diff
@@ -0,0 +1,173 @@
+    wrapper.instance().forceUpdate();
+    expect(handleSearchChange).toHaveBeenCalledTimes(0);
+    expect(wrapper.find(InputGroup).exists()).toBeTruthy();
+    wrapper
+      .find(InputGroup)
+      .find('FormControl')
+      .at(0)
+      .simulate('change', { target: { id: 'close-contact-search-input', value: 'FakeName43' } });
+    expect(handleSearchChange).toHaveBeenCalledTimes(1);
+    expect(handleSearchChange).toHaveBeenCalledWith({ target: { id: 'close-contact-search-input', value: 'FakeName43' } });
+  });
+
+  it('Properly displays an empty table when a monitoree has zero close contacts', () => {
+    const wrapper = getMountedWrapper(mockPatient1, true, ASSIGNED_USERS);
+    expect(wrapper.text()).toContain('No data available in table');
+  });
+
+  it('Properly fills the table with data when a monitoree has more than zero close contacts', () => {
+    const patient = mockPatient2;
+    const closeContactsOfPatient = _.values(mockCloseContacts).filter(x => x.id === patient.id);
+    const wrapper = getMountedWrapper(patient, true, ASSIGNED_USERS);
+    expect(wrapper.find('tbody').find('tr').length).toEqual(closeContactsOfPatient.length);
+    closeContactsOfPatient.forEach(cc => {
+      expect(wrapper.text()).toContain(cc['first_name']);
+      expect(wrapper.text()).toContain(cc['last_name']);
+      expect(wrapper.text()).toContain(formatPhoneNumber(cc['primary_telephone']));
+      expect(wrapper.text()).toContain(cc['email']);
+      expect(wrapper.text()).toContain(Number(cc['contact_attempts']));
+      expect(wrapper.text()).toContain(Number(cc['enrolled_id']));
+      expect(wrapper.text()).toContain(cc['notes']);
+    });
+  });
+
+  // The following tests will throw warning errors. However, they appear to be bugs in the renderer,
+  // and do not impact test functionality
+  it('Properly shows the CloseContact Modal when editing a Close Contact', async () => {
+    const patient = mockPatient2;
+    const wrapper = getMountedWrapper(patient, true, ASSIGNED_USERS);
+    const toggleEditModal = jest.spyOn(wrapper.instance(), 'toggleEditModal');
+    wrapper.instance().forceUpdate();
+    // If the close_contact has a patient_id, the `View Record` option should be present
+    expect(wrapper.find(CloseContactModal).exists()).toBeFalsy();
+    wrapper.find('DropdownToggle').at(0).simulate('click');
+    expect(wrapper.find('DropdownItem').at(0).text()).toContain('Edit');
+    expect(toggleEditModal).toHaveBeenCalledTimes(0);
+    expect(wrapper.state('showEditModal')).toBeFalsy();
+    wrapper.find('DropdownItem').at(0).simulate('click');
+    expect(toggleEditModal).toHaveBeenCalledTimes(1);
+    expect(wrapper.state('showEditModal')).toBeTruthy();
+    expect(wrapper.find(CloseContactModal).exists()).toBeTruthy();
+  });
+
+  it('Properly renders the "View Record" enrollment drop-up option if the Close Contact is enrolled', async () => {
```

Comment:
Can we modify these three tests into two? Properly renders dropdown if CC is enrolled and properly renders dropdown if they are not. that way it's clearer what's happening

Reply:
Fixed after an in-person discussion on the requirements